planet/auth.py

```diff
@@ -80,7 +82,7 @@ def from_env(variable_name: str = None) -> Auth:
             variable_name: Alternate environment variable.
         '''
         variable_name = variable_name or ENV_API_KEY
-        api_key = os.getenv(variable_name)
+        api_key = os.getenv(variable_name, '')
```

Comment:
Resolves

```
planet/auth.py:85: error: Argument 1 to "APIKeyAuth" has incompatible type "Optional[str]"; expected "str"
```

The APIKeyAuth constructor does not take None as an argument (`key: str`) but we might pass it none if there's nothing in the environment.